### PR TITLE
Fix keys example in master [6591]

### PR DESCRIPTION
**Pull request by laura-eprosima**
Backport of #777

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/651/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/565/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/647/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/129/)

File: examples/C++/Keys/keys.cpp [hunk 1]
```diff
@@ -35,25 +35,42 @@ typedef struct
...
+        {
+            n_matched++;
+        }
+        else
+        {
+            n_matched--;
+        }
     }
     int n_matched;
     bool firstConnected;
```
Comment (raquelalvarezbanos) [inline examples/C++/Keys/keys.cpp]: I see #777 added `condition_variable`s whereas this PR doesn't. As a backport I would expect changes here to be exactly the same as in #777, what am I missing?

File: examples/C++/Keys/keys.cpp [hunk 2]
```diff
@@ -301,11 +362,12 @@ void subscriberKeys()
...
     initSubscriber(sampleType, &subListener);
 
-    // wait for the connection
-    while (subListener.n_matched == 0)
+    std::unique_lock<std::mutex> lock(subListener.mutex_);
+    subListener.cv_.wait(lock, [&subListener]()
     {
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    }
+         std::this_thread::sleep_for(std::chrono::milliseconds(100));
```
Comment (raquelalvarezbanos) [inline examples/C++/Keys/keys.cpp]: Is this `sleep_for` needed?

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/685/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/592/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/680/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/157/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/713/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/606/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/708/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/175/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/723/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/616/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/718/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/188/)

**Comment by raquelalvarezbanos** [comment]
Merging, as failures are unrelated.

**Pull request merged**